From 51ad1782df15927491d8bd6ce631de9eb9763a44 Mon Sep 17 00:00:00 2001
From: Evgeny Grin <k2k@narod.ru>
Date: Fri, 19 May 2023 19:26:02 +0300
Subject: [PATCH] Fix unportable '$<' in Makefiles

POSIX specifis '$<' only for "inference" rules (i.e. general rule
like '.c.o:'), while for "target" is undefined.
It is supported as extension for targets by some "make" implementations,
but not all.
The workarounds could be easily used.

Signed-off-by: Karlson2k (Evgeny Grin) <k2k@narod.ru>
---
 doc/man/date-of-man-include.am               | 12 ++++++------
 tests/src/vfs/extfs/helpers-list/Makefile.am |  2 +-
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/doc/man/date-of-man-include.am b/doc/man/date-of-man-include.am
index c6b1e692c..96f9f1049 100644
--- a/doc/man/date-of-man-include.am
+++ b/doc/man/date-of-man-include.am
@@ -12,13 +12,13 @@ MAN_DATE_CMD = \
 	    print POSIX::strftime("$(DATE_FORMAT)", localtime($$fi[9]));' 2>/dev/null
 
 mc.1: $(srcdir)/mc.1.in
-	MAN_FILE=$<; MAN_DATE=$$($(MAN_DATE_CMD)); \
-	sed $(SED_PARAMETERS) $< > $@
+	MAN_FILE='$(srcdir)/mc.1.in'; MAN_DATE=$$($(MAN_DATE_CMD)); \
+	$(SED) $(SED_PARAMETERS) '$(srcdir)/mc.1.in' > '$@'
 
 mcedit.1: $(srcdir)/mcedit.1.in
-	MAN_FILE=$<; MAN_DATE=$$($(MAN_DATE_CMD)); \
-	sed $(SED_PARAMETERS) $< > $@
+	MAN_FILE='$(srcdir)/mcedit.1.in'; MAN_DATE=$$($(MAN_DATE_CMD)); \
+	$(SED) $(SED_PARAMETERS) '$(srcdir)/mcedit.1.in' > '$@'
 
 mcview.1: $(srcdir)/mcview.1.in
-	MAN_FILE=$<; MAN_DATE=$$($(MAN_DATE_CMD)); \
-	sed $(SED_PARAMETERS) $< > $@
+	MAN_FILE='$(srcdir)/mcview.1.in'; MAN_DATE=$$($(MAN_DATE_CMD)); \
+	$(SED) $(SED_PARAMETERS) '$(srcdir)/mcview.1.in' > '$@'
diff --git a/tests/src/vfs/extfs/helpers-list/Makefile.am b/tests/src/vfs/extfs/helpers-list/Makefile.am
index 991a11faa..6c6c04482 100644
--- a/tests/src/vfs/extfs/helpers-list/Makefile.am
+++ b/tests/src/vfs/extfs/helpers-list/Makefile.am
@@ -125,7 +125,7 @@ doc: README.html
 
 # (Thanks to VPATH we don't need to write "$(srcdir)/README". doc/hlp/Makefile.am needlessly does this.)
 README.html: README
-	pandoc --include-in-header=$(srcdir)/README.css.inc -N --old-dashes --toc --toc-depth=4 --standalone -o $@ $<
+	pandoc --include-in-header=$(srcdir)/README.css.inc -N --old-dashes --toc --toc-depth=4 --standalone -o '$@' '$(srcdir)/README'
 
 EXTRA_DIST += README.css.inc
 CLEANFILES += README.html
-- 
2.40.1.windows.1

